pkg/reboot/calculator.go

```diff
@@ -0,0 +1,108 @@
+package reboot
+
+import (
+	"context"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+
+	"github.com/medik8s/self-node-remediation/pkg/utils"
+	"github.com/medik8s/self-node-remediation/pkg/watchdog"
+)
+
+const (
+	MaxTimeForNoPeersResponse = 30 * time.Second
+)
+
+type SafeTimeCalculator interface {
+	GetTimeToAssumeNodeRebooted() time.Duration
+	Start(ctx context.Context) error
+}
+
+type safeTimeCalculator struct {
+	timeToAssumeNodeRebooted, minTimeToAssumeNodeRebooted                   time.Duration
+	wd                                                                      watchdog.Watchdog
+	maxErrorThreshold                                                       int
+	apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout time.Duration
+	log                                                                     logr.Logger
+	k8sClient                                                               client.Client
+	minBatchNumber                                                          int
+}
+
+func NewSafeTimeCalculator(k8sClient client.Client, wd watchdog.Watchdog, maxErrorThreshold int, apiCheckInterval, apiServerTimeout, peerDialTimeout, peerRequestTimeout, timeToAssumeNodeRebooted time.Duration) SafeTimeCalculator {
+	return &safeTimeCalculator{
+		wd:                       wd,
+		maxErrorThreshold:        maxErrorThreshold,
+		apiCheckInterval:         apiCheckInterval,
+		apiServerTimeout:         apiServerTimeout,
+		peerDialTimeout:          peerDialTimeout,
+		peerRequestTimeout:       peerRequestTimeout,
+		timeToAssumeNodeRebooted: timeToAssumeNodeRebooted,
+		k8sClient:                k8sClient,
+		log:                      ctrl.Log.WithName("safe-time-calculator"),
+	}
+}
+
+func (s *safeTimeCalculator) GetTimeToAssumeNodeRebooted() time.Duration {
+	s.calcMinTimeAssumeRebooted()
+	if s.timeToAssumeNodeRebooted < s.minTimeToAssumeNodeRebooted {
+		return s.minTimeToAssumeNodeRebooted
+	}
+	return s.timeToAssumeNodeRebooted
+}
+
+//goland:noinspection GoUnusedParameter
+func (s *safeTimeCalculator) Start(ctx context.Context) error {
+	s.calcMinTimeAssumeRebooted()
+	return nil
+}
+
+func (s *safeTimeCalculator) calcMinTimeAssumeRebooted() {
+	// but the reboot time needs be at least the time we know we need for determining a node issue and trigger the reboot!
```

Comment:
Is the initial "but" intentional?
How about something like "Reboot time needs to be >= to the time needed to determine a node issue and trigger the reboot" ?

Reply:
I'm not sure  IIRC this comment is a copy paste from main, I agree phrasing can be better